packages/api/src/platforms/vtex/utils/shouldUpdateShippingData.ts

```diff
@@ -0,0 +1,156 @@
+}
+
+// Validate if theres a difference between the session geoCoords and orderForm geoCoords
+const checkGeoCoordinates = (
+  address: CheckoutAddress | null | undefined,
+  geoCoordinates: IStoreGeoCoordinates | null | undefined
+) => {
+  return (
+    typeof geoCoordinates?.latitude === 'number' &&
+    typeof geoCoordinates?.longitude === 'number' &&
+    (address?.geoCoordinates[0] !== geoCoordinates?.longitude ||
+      address?.geoCoordinates[1] !== geoCoordinates?.latitude)
+  )
+}
+
+// Validate if the deliveryChannel from the session is different from the selected delivery channel
+// and if so needs to validate if the deliveryChannel for the session is available inside the slas for the item
+const shouldUpdateDeliveryChannel = (
+  logisticsInfo: LogisticsInfo[],
+  session: IStoreSession | null | undefined
+) => {
+  if (!session?.deliveryMode?.deliveryChannel) {
+    return false
+  }
+  const { deliveryChannel } = session.deliveryMode
+  for (const item of logisticsInfo) {
+    if (item.selectedDeliveryChannel !== deliveryChannel) {
+      const matchingSla = item.slas.find(
+        (sla) => sla.deliveryChannel === deliveryChannel
+      )
+      if (matchingSla) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+// Validate if the deliveryMethod from the session is different from the selectedSLA
+// and if so needs to validate if the deliveryMethod for the session is available inside the slas for the item
+const shouldUpdateDeliveryMethod = (
+  logisticsInfo: LogisticsInfo[],
+  session: IStoreSession
+) => {
+  if (!session?.deliveryMode?.deliveryMethod) {
+    return false
+  }
+  const { deliveryMethod } = session.deliveryMode
+  for (const item of logisticsInfo) {
+    if (item.selectedSla !== deliveryMethod) {
+      const matchingSla = item.slas.find((sla) => sla.id === deliveryMethod)
+      if (matchingSla) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+// Validate if the deliveryWindow from the session is different from the deliveryWindow of the SLA
+// and if so needs to validate if the deliveryWindow for the session is available inside the availableDeliveryWindows for the item
+const shouldUpdateDeliveryWindow = (
+  logisticsInfo: LogisticsInfo[],
+  session: IStoreSession
+) => {
+  if (
+    !session?.deliveryMode?.deliveryWindow?.startDate ||
+    !session?.deliveryMode?.deliveryWindow?.endDate
+  ) {
+    return false
+  }
+  const { startDate, endDate } = session.deliveryMode.deliveryWindow
+  for (const item of logisticsInfo) {
+    const { slas } = item
+    for (const sla of slas) {
+      const { availableDeliveryWindows } = sla
+      if (availableDeliveryWindows) {
+        const matchingWindow = availableDeliveryWindows.find(
+          (window) =>
+            window.startDateUtc === startDate && window.endDateUtc === endDate
+        )
+        if (matchingWindow) {
+          return true
+        }
+      }
+    }
```

Comment:
This can be simplified. Also, using `some` instead of `find` is the more appropriate approach in this case, since we don't need to know which deliveryWindow matches the one with the section.

```suggestion
    for (const sla of item.slas) {
      const matchingWindow = sla.availableDeliveryWindows?.some(
        (window) =>
          window.startDateUtc === startDate && window.endDateUtc === endDate
      )
      if (matchingWindow) {
        return true
      }
    }
```

Reply:
Nice!!! Thanks for the contribution ❤️  Amazing!